packages/react-component-library/src/components/CheckboxE/partials/StyledCheckmark.tsx

```diff
@@ -33,12 +42,21 @@ export const StyledCheckmark = styled.div<StyledCheckmarkProps>`
     color: ${color('neutral', 'white')};
   }
 
-  ${StyledCheckbox}:hover &, ${StyledCheckbox}:active & {
-    &::before {
-      box-shadow: 0 0 0 2px ${color('action', '500')};
-      transition: all ${animation('default')};
+  ${({ $hasContainer }) => {
+    if ($hasContainer) {
```

Comment:
There are preferred ways of writing this but did not seem to work as expected (maybe a SC bug) so I arrived at this compromise.